src/recipes.jl

```diff
@@ -287,42 +289,43 @@ function Makie.plot!(gp::GraphPlot)
 
     # plot edges
     edge_plot = edgeplot!(gp, edge_paths;
-                          color=gp.edge_color,
-                          linewidth=gp.edge_width,
-                          gp.edge_attr...)
+        color=prep_attributes(gp.edge_color, @lift(getedgeinds($graph, $(gp.edge_color))), dfth.edge_color),
```

Comment:
Funcionally I am on board but I am not super happy with this construct. Because a change of `gp.edge_color` from the outside would trigger two things:
- recalculation of observable definition within `prep_attributes`
- recalculation of `getedgeinsd` output which then triggers recalculation of `prep_attributes`

Havn't though this through, but maybe it is worth splitting `prep_attributes` in 
```
prep_node_attributes(attr::Observable, graph::Observable{g}, default)
``` 
and 
```
prep_edge_attributes(attr::Observable, graph::Observable{g}, default)
    @lift begin
        if issingleattribute($attr)
            isnothing($attr) ? $default : $attr
        elseif $attr isa AbstractVector
            $attr
        else
            [getattr($attr, i, $default) for i in getedgeinds($graph, $attr)]
        end
    end
end
```
So `@lift getedgeinds(..)` for edges and `@lift vertice(g..)` could become part of the `prep_attributes` lift.

Reply:
There are two subproblems:
- even if `gp.edge_color` is an vector, the `tempobs` will be updated every single time
- the observable created in `prep_attributes` lifts both on `gp.edge_color` and on the `tempobs`. Therefore, change of `gp.edge_color` will trigger `prep_attributes`, then `tempobs` and then `tempobs` will trigger `prep_attributes` again.

I think this function
```julia
prep_edge_attributes(attr::Observable, graph::Observable{g}, default)
    @lift begin
        if issingleattribute($attr)
            isnothing($attr) ? $default : $attr
        elseif $attr isa AbstractVector
            $attr
        else
            [getattr($attr, i, $default) for i in getedgeinds($graph, $attr)]
        end
    end
end
```
elides both of those problems. I think withint `prep_edge_attributes` it still makes perfect sense to use the trait dispatches function `getedgeinds`. But we get rid of one layer of observables.

Reply:
You are right! Nice catch! I though there was a dependency tree that decided what/how to execute with `Observables`.
However the following code which is similar indeed is executed twice:
```julia
using Observables

edge_color = Observable(1)
graph = Observable(2)

tempobs = map(edge_color, graph) do e,g
    @info("tempobs update")
    e + g
end

prep_attrs = map(edge_color, tempobs) do e, t
    @info("prep_attrs update")
    e + t
end

myplot = map(prep_attrs) do p
    @info("myplot update")
    p * 10
end
```

Changing edge_color[] will execute prepr_attrs twice:
```julia-repl
julia> edge_color[] = 3;
[ Info: tempobs update
[ Info: prep_attrs update
[ Info: myplot update
[ Info: prep_attrs update
[ Info: myplot update
```

I will rewrite it as you suggested together with the other comments. 
Also in order to keep it symmetric I would also make a `prep_vertex_attributes` that also gets a graph `Observable`.